system/masking/plugins/datamasker/readme.md

```diff
@@ -0,0 +1,182 @@
+- **memberOf/@name** is a required user-defined set label.
+
+rule instances will frequently be specific to a content format. For example, a rule applied to XML markup value may include dependencies on XML syntax which are not applicable when masking JSON content. Rules specific to a markup format can be associated with that format using **@contentType**. Requests to mask content of a type will apply all rules without a **@contentType** value or with a matching value, and will exclude all rules with a non-matching value.
+
+#### CSerialTokenRule
+
+An extension of [CRule](#crule), this identifies content substrings to be masked based on matching start and end tokens in the content buffer. For each occurrence of a configured start token that is balanced by a corresponding configured end token, the characters between the tokens are masked.
+
+This class may be used with [TSerialProfile](#tserialprofile).
+
+Configuration includes:
+
+- **@endToken** is an optional character sequence expected to immediately follow a content substring to be masked. This might be an XML element end tag, or a terminating double quote for a JSON value. A newline, i.e., *\\n*, is assumed if omitted or empty, for masking values such as HTTP headers.
+- **@matchCase** is an optional Boolean flag controlling whether token matches are case sensitive (*true*) or insensitivie (*false*). Matches are case insensitive by default. The implementation is based on ASCII data; case sensitive comparisons of similarly encoded non-ASCI text can work, but case insensitive comparisons are not supported.
+- **@startToken** is a required character sequence expected to precede a content substring to be masked.
+
+> No content type knowledge is implied by this class. An instance with **@contentType** of *xml* does not inherently know how to find values in XML markup. The defined tokens must include characters such as `<` and `>` to match an element name and quotes to match attribute values.
+
+#### TPlugin
+
+An implementation of `IDataMaskingProfileIterator` used for transforming a configuration property tree into a collection of profiles to be returned by a library entry point function. Created profiles are of the same class, identified by the template parameter **profile_t**.
+
+Configuration includes:
+
+- **profile** An optional and repeating element defining a profile. The content of this element depends on **profile_t**.
+- **...** If and only if **profile** is not included as a child of the configuration, the configuration node itself will be viewed as a configursation for **profile_t**. The name of the configuration node is not relevant in this case; it is assumed to be a node recognized by the host platform's `IDataMaskingEngine` implementation as a plugin configuration node.
```

Comment:
No. The given property tree may contain a list of `profile` elements, which is the first bullet point. If the given property tree contains no `profile` elements, it is assumed to be equivalent to a single `profile` element (equivalent because the name is not relevant). I understand the confusion, but I'm not how to better express this.